src/settings-ui/Settings.UI.Library/SettingsBackupAndRestoreUtils.cs

```diff
@@ -0,0 +1,965 @@
+                        Logger.LogInfo($"BackupSettings, {currentFile.Value} does not exists.");
+                        doBackup = true;
+                    }
+
+                    if (doBackup)
+                    {
+                        // add to list of files we noted as needing backup
+                        updatedSettingsFiles.Add(currentFile.Key, currentFile.Value);
+
+                        // mark overall flag that a backup will be made
+                        anyFileBackedUp = true;
+
+                        // write the export version of the settings file to backup location.
+                        var relativePath = currentFile.Value.Substring(appBasePath.Length + 1);
+                        var backupFullPath = Path.Combine(fullBackupDir, relativePath);
+
+                        TryCreateDirectory(fullBackupDir);
+                        TryCreateDirectory(Path.GetDirectoryName(backupFullPath));
+
+                        Logger.LogInfo($"BackupSettings writing, {backupFullPath}, dryRun:{dryRun}.");
+                        if (!dryRun)
+                        {
+                            File.WriteAllText(backupFullPath, currentSettingsFileToBackup);
+                        }
+                    }
+                    else
+                    {
+                        // if we found no reason to backup this settings file, record that in this collection
+                        skippedSettingsFiles.Add(currentFile.Key, (currentFile.Value, currentSettingsFileToBackup));
+                    }
+                }
+
+                if (!anyFileBackedUp)
+                {
+                    // nothing was done!
+                    return (false, $"BackupAndRestore_NothingToBackup", "Informational");
+                }
+
+                // add skipped.
+                foreach (var currentFile in skippedSettingsFiles)
+                {
+                    // if we did do a backup, we need to copy in all the settings files we skipped so the backup is complete.
+                    // this is needed since we might use the backup on another machine/
+                    var relativePath = currentFile.Value.path.Substring(appBasePath.Length + 1);
+                    var backupFullPath = Path.Combine(fullBackupDir, relativePath);
+
+                    Logger.LogInfo($"BackupSettings writing, {backupFullPath}, dryRun:{dryRun}");
+                    if (!dryRun)
+                    {
+                        TryCreateDirectory(fullBackupDir);
+                        TryCreateDirectory(Path.GetDirectoryName(backupFullPath));
+
+                        File.WriteAllText(backupFullPath, currentFile.Value.settings);
+                    }
+                }
+
+                // add manifest
+                var manifestData = new
+                {
+                    CreateDateTime = DateTime.UtcNow.ToString("u", CultureInfo.InvariantCulture),
+                    @Version = Helper.GetProductVersion(),
```

Comment:
@jefflord
Should we show this in the backup information? Not sure if this is an interesting information.

Reply:
I just noticed this, and it's a good question. I am not against this, but really it's better if it's something never needed.